service/internal/checksum/middleware_compute_input_checksum.go

```diff
@@ -65,6 +65,10 @@ type ComputeInputPayloadChecksum struct {
 	// when used with trailing checksums, and aws-chunked content-encoding.
 	EnableDecodedContentLengthHeader bool
 
+	checksum string
+
+	sha256Checksum string
```

Comment:
just for my education, why is sha256 treated differently than other algorithms for checksum?

Reply:
Yes, sha256 value is cached in ctx [here](https://github.com/aws/aws-sdk-go-v2/blob/fcd6c667064dfd76542078bcd5b435e9df401aef/service/internal/checksum/middleware_compute_input_checksum.go#L211) for later sigv4